### DISCUSSION TITLE
Why don't we have 'atom-languageclient' package on repository?

### DISCUSSION THREAD

**Discussion by leotada** [pulsar-edit/.github#243] [Community Package Help]
Why don't we have 'atom-languageclient' package on repository?
This is not a important package for LSP usage? or you are using other alternative?

**Comment by Daeraxa** [comment]
I think there are a few things to this:
1) If it was on atom.io I would have expected it to have been migrated to the PPR but I can't seem to find it on there.
2) It might be that atom-community no longer support it and I'm not sure they have ever pushed any packages to the PPR so it is possibly unmaintained
3) One of the team has been working on a `generic-lsp` package which might well become the new standard implementation for a generic LSP client for Pulsar

**Reply by leotada** [reply]
Where we can follow the next development efforts like this?

**Comment by savetheclocktower** [comment]
The `atom-languageclient` package is designed to be a dependency for IDE backend packages to use, not as something that users install directly. That's why it's published on NPM instead of the Pulsar package repository.

The good news is that there are _lots_ of Pulsar packages that use `atom-languageclient` in this manner. [Any package whose name is of the form `ide-[language]`](https://web.pulsar-edit.dev/packages/search?q=ide-) and has a description like “____ language support“ is going to be such a package. Install the ones that make sense for the languages you use.

Some of the LSP support needs updating, so I've been working on [my own fork of `atom-languageclient`](https://github.com/savetheclocktower/atom-languageclient) and am using it in a few experimental packages. (Right now [`pulsar-ide-typescript-alpha`](https://web.pulsar-edit.dev/packages/pulsar-ide-typescript-alpha) and [`pulsar-ide-ruby-solargraph`](https://web.pulsar-edit.dev/packages/pulsar-ide-ruby-solargraph) are the only ones I've published.)

Finally, [`generic-lsp`](https://github.com/mauricioszabo/generic-lsp) is another good solution, especially if you use an obscure language server and there isn't a dedicated package for it. My efforts will go toward `atom-languageclient` because `generic-lsp` is written in ClojureScript and I'm not about to learn all that stuff! But it's good to have options.
